Mathlib/Algebra/Field/Opposite.lean

```diff
@@ -35,44 +33,46 @@ theorem unop_ratCast [RatCast α] (q : ℚ) : unop (q : αᵐᵒᵖ) = q :=
 #align mul_opposite.unop_rat_cast MulOpposite.unop_ratCast
 #align add_opposite.unop_rat_cast AddOpposite.unop_ratCast
 
-variable (α)
+instance instDivisionSemiring [DivisionSemiring α] : DivisionSemiring αᵐᵒᵖ where
+  toSemiring := instSemiring
+  __ := instGroupWithZero
 
-instance instDivisionSemiring [DivisionSemiring α] : DivisionSemiring αᵐᵒᵖ :=
-  { MulOpposite.instGroupWithZero α, MulOpposite.instSemiring α with }
+instance instDivisionRing [DivisionRing α] : DivisionRing αᵐᵒᵖ where
+  toRing := instRing
```

Comment:
```suggestion
  __ := instRing
```

Reply:
(and throughout)

Reply:
Why should I do this? It's more prone to people accidentally rearranging sources and messing up the instance.

Reply:
I think that's a problem for a linter to solve, not something that we want in the source code. This spelling is more brittle than using `__` everywhere, and still doesn't stop you writing `toTheGrandparentClass :=` and still making a mess.

Reply:
Also note that despite you "fixing the mess" of some of the parents orders, this seems not to have actually impacted performance.

Reply:
I suspect this is more due to the relatively little use of `MulOpposite` than poor performance of my technique

Reply:
Oh, I don't claim your technique is problematic performance-wise; I think it's uglier and more fragile than my suggested technique, while being exactly the same performance-wise.